path_login.go

```diff
@@ -618,9 +625,14 @@ func (c *additionalClaims) verifyVM(vmName string) error {
 }
 
 // verifyVMSS checks the additional claims in the token against
-// the provided vm_name field on login
+// the provided vm_name field on login. We have to check both the uniform and flexible
+// claim patterns.
```

Comment:
Minor suggestion to provide some reasoning around why we try the flexible claim pattern first (which makes sense given it's the recommended and default orchestration option from Microsoft)

```suggestion
// the provided vm_name field on login. We have to check both the
// uniform and flexible claim patterns. Since flexible is recommended
// and default (https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-orchestration-modes), check it first.
// If flexible fails, fall back to uniform.
```